systemtests/src/test/java/io/enmasse/systemtest/shared/standard/QueueTest.java

```diff
@@ -96,6 +97,45 @@ public static void runQueueTest(AmqpClient client, Address dest, int countMessag
         assertThat("Wrong count of messages received", actual, is(msgs.size()));
     }
 
+    public static void runQueueTestRetrying(AmqpClient client, Address dest, int countMessages, Duration retryTimeout) throws InterruptedException, TimeoutException, ExecutionException, IOException {
+        List<String> msgs = TestUtils.generateMessages(countMessages);
+        long timeoutMs = countMessages * ClientUtils.ESTIMATE_MAX_MS_PER_MESSAGE;
+
+        TestUtils.waitUntilConditionOrFail(() -> {
+            log.info("Start sending with " + timeoutMs + " ms timeout");
+            Count<Message> predicate = new Count<>(msgs.size());
+            Future<Integer> numSent = client.sendMessages(dest.getSpec().getAddress(), msgs, predicate);
```

Comment:
If you succeed in `numSent > 0 ` and `numSent < msg.length` the send loop will retry.  This will mean that the number of messages of the queue ends up being greater than msg.length, which will make for an odd looking test failure.   You could reduce msg by actual at line 112 to avoid that.